### PR TITLE
For Comment: Adding JsonPath attribute extractor CQL Function

### PR DESCRIPTION
**Pull request by aheyne**
This adds the CQL function 'jsonattr' to allow specifying jsonpaths in a CQL filter function, in particular, through WFS. This allows queries like these, and more, to work. See the unit test for more examples.

`jsonattr('$.data.[FooBar]') = 'baz'`
`jsonattr('$.data.[Foo Bar]') = 'baz'`
`jsonattr('$.data.[Foo.Bar]') = 'baz'`

Any thoughts on a better way to do this? The JsonPath escape in FastPropertyName is necessary because it is called when using 
`val path = getExpression(0).evaluate(null).asInstanceOf[String]`
to extract the function parameter (the passed in path), otherwise FastPropertyName will throw exceptions.

### CODE REVIEW

**Comment by elahrvivaz** [comment]
can we change the function name to `jsonPath`?

File: geomesa-features/geomesa-feature-kryo/pom.xml [hunk 1]
```diff
@@ -39,11 +39,11 @@
...
-        <dependency>
-            <groupId>org.locationtech.geomesa</groupId>
-            <artifactId>geomesa-security_${scala.binary.version}</artifactId>
-            <scope>test</scope>
-        </dependency>
+        <!--<dependency>-->
+            <!--<groupId>org.locationtech.geomesa</groupId>-->
+            <!--<artifactId>geomesa-security_${scala.binary.version}</artifactId>-->
+            <!--<scope>test</scope>-->
+        <!--</dependency>-->
```
Comment (elahrvivaz) [inline geomesa-features/geomesa-feature-kryo/pom.xml]: any reason for removing this? we're going to need it to work with security on the classpath

**Comment by aheyne** [comment]
Yeah, I was planning on changing it to something like that, jsonPath it is.

File: geomesa-features/geomesa-feature-kryo/pom.xml (hunk 1 above)
Comment (aheyne) [inline geomesa-features/geomesa-feature-kryo/pom.xml]: I don't think so, I just haven't cleaned this up yet.

File: geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonAttrFunction.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.geotools.filter.capability.FunctionNameImpl
+import org.geotools.filter.capability.FunctionNameImpl.parameter
+import org.geotools.filter.expression.PropertyAccessors
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeature
+import org.opengis.filter.expression.{PropertyName, VolatileFunction}
+
+class JsonAttrFunction extends FunctionExpressionImpl(
+  new FunctionNameImpl("jsonattr",
+    parameter("propertyName", classOf[PropertyName]),
```
Comment (jnh5y) [inline geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonAttrFunction.scala]: Oh snap... @elahrvivaz is right.  If this is a 'string', we can save on the changes to the FastPropertyName. 

*headdesk*

Comment (aheyne) [inline geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/json/JsonAttrFunction.scala]: Oohhh

**Pull request closed without merging**